src/main/webapp/index.html

```diff
@@ -35,39 +35,42 @@ <h1 class="index-registration-block-child">TravIS: Travel Information System</h1
         <form action="/calculate-trip" method="POST" id="startTripForm">
           <div class="form-row">
             <div class="form-group col-md-3">
-              <input type="text" placeholder="Destination" class="form-control" 
-                name="inputDestination" id="inputDestination">
+              <label for="inputDestination">Destination</label>
+              <input type="text" placeholder="Where to?" class="form-control" 
+                name="inputDestination" id="inputDestination" 
+                onchange="checkValidInput('inputDestination'); checkNextButton();
+                  checkSubmitButton()">
             </div>
             <div class="form-group col-md-3">
-              <input type="text" placeholder="Start Date" class="form-control" 
-                name="inputStartDate" id="inputStartDate">
+              <label for="inputStartDate">Start Date</label>
+              <input type="date" class="form-control" name="inputStartDate" 
+                id="inputStartDate" oninput="setEndDateValue()"
+                onchange="checkValidInput('inputStartDate'); checkNextButton(); 
+                  checkSubmitButton()">
             </div>
             <div class="form-group col-md-3">
-              <input type="text" placeholder="End Date" class="form-control is-valid" 
-                name="inputEndDate" id="inputEndDate" readonly>
+              <label for="inputEndDate">End Date</label>
+              <input type="date" class="form-control" name="inputEndDate" 
+                id="inputEndDate" readonly>
             </div>
             <div class="form-group">
-              <button type="button" class="btn btn-secondary">Next</button>
+              <input type="button" class="btn btn-success" value="Next" 
+                id="toggle-stage-button" onclick="toggleStartTripInputStage()">
             </div>
           </div>
-          <div class="form-row">
+          <div class="form-row" id="add-pois-container">
             <div class="form-group col-md-3">
-              <input type="text" placeholder="Point of Interest" 
-                class="form-control" id="inputPoi">
-            </div>
-            <div class="form-group">
-              <button type="button" class="btn btn-primary">Add POIs</button>
-            </div>
-            <div class="form-group">
-              <input type="button" value="California (click to remove)" 
-                class="btn btn-secondary poi-input" name="California">
+              <input type="text" placeholder="Point of Interest" id="inputPoi" 
+                class="form-control" onchange="checkValidInput('inputPoi'); 
```

Comment:
Optional: So for me the workflow was to type the POI i'm interested in then click elsewhere on the page, then click on the add button. I'd prefer to just have the add button become clickable once I've typed enough characters, so that I can skip that first click. If I'm understanding correctly this is because the onchange event is only issued when the input box changes focus (ie, when I click away).

This is a small issue, and I suspect some of this UI will change once you start integrating the maps POI suggestion API. So probably easiest to deal with this later.

Reply:
Yes, I was aware of this -- you're right, this aspect of the UI changes once the Google Place autofill is added, so I'm going to leave this as-is in this CL. Thanks for the note though!